src/main/java/io/codearte/jfairy/producer/company/CompanyProvider.java

```diff
@@ -15,30 +16,68 @@
 	private static final String COMPANY_SUFFIX = "companySuffixes";
 	private static final String COMPANY_EMAIL = "companyEmails";
 
+	private String name;
+	private String domain;
+	private String email;
+	private String vatIdentificationNumber;
+
 	private BaseProducer baseProducer;
 	private DataMaster dataMaster;
 
 	private final VATIdentificationNumberProvider vatIdentificationNumberProvider;
 
 	@Inject
-	public CompanyProvider(BaseProducer baseProducer, DataMaster dataMaster, VATIdentificationNumberProvider vatIdentificationNumberProvider) {
+	public CompanyProvider(BaseProducer baseProducer,
+						   DataMaster dataMaster,
+						   VATIdentificationNumberProvider vatIdentificationNumberProvider,
+						   @Assisted CompanyProperties.CompanyProperty... companyProperties) {
 		this.baseProducer = baseProducer;
 		this.dataMaster = dataMaster;
 		this.vatIdentificationNumberProvider = vatIdentificationNumberProvider;
+
+		for (CompanyProperties.CompanyProperty companyProperty : companyProperties) {
+			companyProperty.apply(this);
+		}
 	}
 
 
 	@Override
 	public Company get() {
 
```

Comment:
@OlgaMaciaszek
I did not commit the suggested changes mentioned in the above comment. Should I do them and do another pull requests?